src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts

```diff
@@ -0,0 +1,98 @@
+import { ChartType, Plugin } from "chart.js";
+import { chartFontColor } from "../../../../helpers/figures/charts/chart_common";
+import { Color } from "../../../../types";
+
+interface ChartShowValuesPluginOptions {
+  showValues: boolean;
+  background?: Color;
+  horizontal?: boolean;
+  callback?: (any) => any;
+}
+
+declare module "chart.js" {
+  interface PluginOptionsByType<TType extends ChartType> {
+    chartShowValuesPlugin?: ChartShowValuesPluginOptions;
+  }
+}
+
+/** This is a chartJS plugin that will draw connector lines between the bars of a Waterfall chart */
+export const chartShowValuesPlugin: Plugin = {
+  id: "chartShowValuesPlugin",
+  afterDatasetsDraw(chart: any, args, options: ChartShowValuesPluginOptions) {
+    if (!options.showValues) {
+      return;
+    }
+    const drawData = chart._metasets?.[0]?.data;
+    if (!drawData) {
+      return;
+    }
+    const ctx = chart.ctx;
+    ctx.save();
+
+    ctx.textAlign = "center";
+    ctx.textBaseline = "middle";
+    ctx.fillStyle = chartFontColor(options.background);
+    ctx.strokeStyle = chartFontColor(ctx.fillStyle);
+
+    chart._metasets.forEach(function (dataset) {
+      switch (dataset.type) {
+        case "pie": {
+          for (let i = 0; i < dataset._parsed.length; i++) {
+            const bar = dataset.data[i];
+            const { startAngle, endAngle, innerRadius, outerRadius } = bar;
+            const midAngle = (startAngle + endAngle) / 2;
+            const midRadius = (innerRadius + outerRadius) / 2;
+            const x = bar.x + midRadius * Math.cos(midAngle);
+            const y = bar.y + midRadius * Math.sin(midAngle) + 7;
+
+            ctx.fillStyle = chartFontColor(bar.options.backgroundColor);
+            ctx.strokeStyle = chartFontColor(ctx.fillStyle);
+
+            let value = dataset._parsed[i];
+            if (options.callback) {
+              value = options.callback(value);
+            }
+            ctx.strokeText(value, x, y);
+            ctx.fillText(value, x, y);
+          }
+          break;
+        }
+        case "bar":
+        case "line": {
+          const yOffset = dataset.type === "bar" && !options.horizontal ? 0 : 3;
+          for (let i = 0; i < dataset._parsed.length; i++) {
+            const point = dataset.data[i];
+            const value = options.horizontal ? dataset._parsed[i].x : dataset._parsed[i].y;
+            const displayedValue = options.callback ? options.callback(value) : value;
```

Comment:
The displayed points don't take the dataset format. Is it intentional ?
![image](https://github.com/odoo/o-spreadsheet/assets/31045223/d6a53af2-dbf9-4ffe-87eb-463b02752208)